streambed-confidant/examples/main.rs

```diff
@@ -0,0 +1,79 @@
+use std::error::Error;
+
+use clap::Parser;
+use git_version::git_version;
+use log::info;
+use streambed::secret_store::SecretStore;
+use streambed_confidant::{args::SsArgs, FileSecretStore};
+
+/// Service CLI that brings in the args we need for configuring access to a secret store
+#[derive(Parser, Debug)]
+#[clap(author, about, long_about = None, version=git_version!())]
+struct Args {
+    #[clap(flatten)]
+    ss_args: SsArgs,
+}
+
+#[tokio::main]
+async fn main() -> Result<(), Box<dyn Error>> {
+    let args = Args::parse();
+
+    env_logger::init();
+    info!("Confidant authenticator started");
+
+    // Setup and authenticate our service with the secret store
+
+    let line = streambed::read_line_from_stdin().await.unwrap();
+    assert!(!line.is_empty(), "Cannot source a line from stdin");
```

Comment:
This assert will never be hit as `read_line_from_stdin` doesn't time out

Reply:
The read line will return once it encounters a newline. Thus, if you just hit return without piping stdin then the assert comes into play.

Reply:
Right, of course, that makes sense...

Reply:
Yep